File: hydro.py
#!/usr/bin/env python
# hydrostatics results analysis with NumPy & matplotlib
# D. DJokic, Mar-2013
# calculates only drafts - can be extended further


import numpy as np 
import matplotlib.pyplot as plt 

laun_draft=float(input("Input Draft to check Displacement [m]: "))
filename=raw_input("Enter CSV filename: ")

# reading data from CSV file - edit to suit data range

hyd=np.loadtxt(filename, dtype=float, comments="#", delimiter=',', converters=None, skiprows=2, usecols=(0,1,3), unpack=False, ndmin=0)
records=len(hyd)

T=[]  		#draft moulded [m]
Tk=[] 		#draft to bottom of the keel [m]
Displ=[]	#displacement [t]	

for i in range (0, records):
	T.append(hyd[i,0])
	Tk.append(hyd[i,1])
	Displ.append(hyd[i,2])
print(Displ)

# fitting to 2nd degree curve
p=np.polyfit(T, Displ, 2)
Calc_Dis=p[0]*laun_draft**2+p[1]*laun_draft+p[2]
print("\n")
print(Calc_Dis)

# display grid
fig=plt.figure()
ax=fig.add_subplot(111)

plt.plot(T, Displ)
plt.xlabel("Draft, moulded [m]")
plt.ylabel("Displacement [t]")
ax.grid()
plt.show()

c_disp=[]
drft=[]
for draft in range(0,20):
	pcal=p[0]*draft**2+p[1]*draft+p[2]
	c_disp.append(pcal)
	drft.append(draft)
plt.plot(drft, c_disp)
plt.xlabel("Draft, moulden [m]")
plt.ylabel("Displacement-p [t]")
ax.grid()
plt.show()

File: hydro.m
#!/usr/bin/env octave -q
# hydrostatic results analytics
# units metrics
# D. Djokic, 24-Mar-2013

# draft, moulded = T [m]
# draft below keel =Tk[m]
# calculates only drafts - can be extended further

filename=input("Text Filename with data: ", 's');

# reading data from CSV file - edit to suit data range

T=dlmread(filename, ',', 'A3:A151');
Tk=dlmread(filename, ',', 'B3:B151');
VOLMLD=dlmread(filename, ',', 'C3:C151');
DISPSW=dlmread(filename, ',', 'D3:D151');
LCB=dlmread(filename, ',', 'E3:E151');
TCB=dlmread(filename, ',', 'F3:F151');
VCB=dlmread(filename, ',', 'G3:G151');
LCF=dlmread(filename, ',', 'H3:H151');
WLA=dlmread(filename, ',', 'I3:I151');
TPC=dlmread(filename, ',', 'J3:J151');
KML=dlmread(filename, ',', 'K3:K151');
KMT=dlmread(filename, ',', 'L3:L151');

# graph of displacement depending on Tk
plot(Tk, DISPSW);
xlabel('Tk [m]');
ylabel('DISPSW [t]');
title('BassDrill Beta - DISPSW(Tk)');
print('BDB-displacement.dxf', '-ddxf');
print('BDB-displacement.png', '-dpng');

# launching case - draft below 7.8m ('hump' on curve)

Tk1=dlmread(filename, ',','B3:B44');
DISPSW1=dlmread(filename, ',', 'D3:D44');
plot(Tk1, DISPSW1);
xlabel('Tk [m]');
ylabel('DISPSW [t]');
title('BassDrill Beta - DISPSW(Tk)-Launching');
print('BDB-displacementL.dxf', '-ddxf');
print('BDB-displacementL.png', '-dpng');

# fitting curve - 1st degree
p=polyfit(Tk1, DISPSW1, 1);

# assembling curve & calculating weight on launching, based on draft
draft=input('Input launching draft [m]: ');
displ=p(1)*draft+p(2)